physx/src/physics/error_callback.rs

```diff
@@ -0,0 +1,42 @@
+use std::ffi::{c_void, CStr};
+
+use enumflags2::BitFlags;
+use physx_sys::{create_error_callback, PxErrorCallback};
+
+use crate::foundation::ErrorCode;
+
+pub trait ErrorCallback: Sized {
+    fn report_error(&self, code: BitFlags<ErrorCode>, message: &str, file: &str, line: u32);
+    /// # Safety
+    ///
+    /// Do not override this method.
+    unsafe fn into_px(self) -> *mut PxErrorCallback {
+        unsafe extern "C" fn on_message_shim<L: ErrorCallback>(
+            code: physx_sys::PxErrorCode::Enum,
+            message: *const i8,
+            file: *const i8,
+            line: u32,
+            this: *const c_void,
+        ) {
+            let this = &*this.cast::<L>();
+            let msg = CStr::from_ptr(message.cast());
+            let msg = msg.to_str().unwrap(); // SAFETY: We're dealing with basic char* here, so should always convert cleanly to str.
```

Comment:
not if it doesn't contain valid utf8?

Reply:
I'm making two assumptions to say that isn't an issue: All data passed into PhysX from this library is already UTF-8 so it cannot be broken, and all formatted strings in PhysX should contain only ASCII. If we can find a nice way to handle this without silently ignoring errors I'm fine with being stricter/more careful.

Reply:
Yeah, it probably won't ever be an issue.

I guess you could imagine a situation where someone for some reason decide to compile physx in a directory containing non-ascii characters, on linux, with the system encoding set to something other than utf8. Then `__FILE__` could contain non-ascii chars, which would end up in `file` :)

Reply:
but perhaps at least change the SAFETY comment? "basic char*" isn't really a guarantee for anything

Reply:
Switched to `to_string_lossy` unconditionally, PhysX doesn't log enough for it to matter.